File: dynajoy/launch/robot_launch.py
import os
from launch import LaunchDescription
from launch_ros.actions import Node
from ament_index_python.packages import get_package_share_directory

deadzone    = 0.1
sensitivity = 1.0

# robot_params.yaml is installed alongside the package (see setup.py)
_params_file = os.path.join(
    get_package_share_directory('dynajoy'),
    'config',
    'robot_params.yaml',
)


def generate_launch_description():
    return LaunchDescription([

        # ── Joystick driver ───────────────────────────────────────────────────
        Node(
            package='joy_linux',
            executable='joy_linux_node',
            name='joy_linux_node',
            parameters=[
                {'controller_id': 0},
                {'deadzone': deadzone},
                {'sensitivity': sensitivity},
            ],
            remappings=[('/joy', '/controller1/joy')],
        ),

        # ── Arm controller ────────────────────────────────────────────────────
        # robot_params.yaml is loaded via the ROS 2 parameter server.
        # All values are visible and changeable at runtime:
        #   ros2 param list /command_publisher
        #   ros2 param set  /command_publisher ik.step_mm 8.0
        Node(
            package='dynajoy',
            executable='command_publisher',
            name='command_publisher',
            parameters=[
                _params_file,                       # ← robot_params.yaml
                {'controller_id': 1},
                {'deadzone': deadzone},
                {'sensitivity': sensitivity},
            ],
            remappings=[('/joy', '/controller1/joy')],
        ),
    ])

File: dynamixel_control_cpp/launch/robot_launch.cpp
#include <memory>
#include "launch/launch.hpp"
#include "launch_ros/actions/node.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace launch;
using namespace launch_ros;

LaunchDescription generate_launch_description() {
    return LaunchDescription({
        Node(
            "joy_linux_node",
            "joy_linux",
            {
                {"controller_id", 0},
                {"deadzone", 0.1},
                {"sensitivity", 1.0}
            },
            {
                {"/joy", "/controller1/joy"}
            }
        ),
        Node(
            "command_publisher",
            "dynamixel_control_cpp",
            {
                {"controller_id", 1},
                {"deadzone", 0.1},
                {"sensitivity", 1.0}
            },
            {
                {"/joy", "/controller1/joy"}
            }
        )
    });
}